### PR TITLE
Improve `Day` component

### PR DESCRIPTION
**Pull request by PeterMK85**
### Description

+ Isolate `Day` render related props
+ Replace `MouseMove` with `MouseEnter`
+ Remove unnecessary binding

### Review

- [x] Verify that all classes affected by the changes have **good** top-level documentation.
- [x] Verify that new code has been covered with specs (and features if applicable).
- [x] Verify that a corresponding meaningful changelog entry has been added (`* Add Github pull request template`), for this change if applicable.
- [x] Test manually.
- [x] Get one :+1: from code review.

### Pre-merge checklist

- [x] The PR relates to a single subject with a clear title and description in grammatically correct, complete sentences.
- [x] Verify that all new dependencies are included in `package.json`.
- [x] Double check the quality of [commit messages](http://chris.beams.io/posts/git-commit/).
- [x] Squash related commits together.

### CODE REVIEW

File: src/calendar/day.jsx [hunk 1]
```diff
@@ -1,53 +1,69 @@
...
-
-  _onMouseMove(e) {
-    e.preventDefault()
-    const {date, onMouseMove} = this.props
-    if (onMouseMove) {
-      onMouseMove(date)
-    }
-  }
+const Day = ({
+  blockClassName,
```
Comment (PeterMK85) [inline src/calendar/day.jsx]: For a future enhancement:
this approach its easier to apply a render prop approach

File: src/calendar/day.jsx [hunk 2]
```diff
@@ -1,53 +1,69 @@
...
+      'is-prev_month': isMonthPrev,
+      'is-selectable': isSelectable,
+      'is-selected': isSelected,
+      'is-start_selection': isSelectionStart,
+      'is-today': isToday,
+      'is-weekend': isWeekend,
+      'is-working_day': isWorkday,
+    })}
+    onClick={handleOnClick}
+    onMouseEnter={handleOnEnter}
```
Comment (PeterMK85) [inline src/calendar/day.jsx]: Previously it was a `mouseMove` I think its enough to fire the event on `mouseEnter`

File: css/date_picker.styl [hunk 3]
```diff
@@ -97,10 +95,13 @@ calendarWidth = 250px
     background #ffffff
     position relative
     border-radius 50%
+    margin 0
```
Comment (PeterMK85) [inline css/date_picker.styl]: Needed for `button` element because currently we are not importing any `reset.css`

File: src/calendar/week.jsx [hunk 4]
```diff
@@ -38,19 +33,14 @@ export default class Week extends React.Component {
...
     onDisabledDayClick: PropTypes.func.isRequired,
     selectedMax: datePropType,
     selectedMin: datePropType,
     today: datePropType.isRequired,
     weekStartsOn: PropTypes.oneOf(DAYS_IN_WEEK),
   }
 
-  static defaultProps = {
-    data: {},
-    blockClassName: BLOCK_CLASS_NAME,
```
Comment (PeterMK85) [inline src/calendar/week.jsx]: Not required/not used

File: src/calendar/week.jsx [hunk 5]
```diff
@@ -71,7 +61,7 @@ export default class Week extends React.Component {
 
   _dateSelected(date) {
     const {selectedMin, selectedMax} = this.props
-    return (
+    return Boolean(
```
Comment (PeterMK85) [inline src/calendar/week.jsx]: To always return a boolean

File: src/calendar/week.jsx [hunk 6]
```diff
@@ -101,54 +91,54 @@ export default class Week extends React.Component {
     return false
   }
 
-  _dateClasses(date) {
```
Comment (PeterMK85) [inline src/calendar/week.jsx]: added as a direct prop resolve value

File: css/base_style.styl [hunk 7]
```diff
@@ -24,15 +24,19 @@
     opacity .5
 
   &-day
+    margin 0
```
Comment (PeterMK85) [inline css/base_style.styl]: Needed for `button` element because currently we are not importing any `reset.css`

**Comment by PeterMK85** [comment]
Someone? Anyone?

**Comment by havenchyk** [comment]
@PeterMK85 please, be patient, I'll take a look when I have some time

File: src/calendar/__specs__/day_spec.js [hunk 8]
```diff
@@ -1,97 +1,179 @@
...
+  isNonSelectable: false,
+  isSelectable: false,
+  isSelected: false,
+  isSelectionEnd: false,
+  isSelectionStart: false,
+  isToday: false,
+  isWeekend: false,
+  isWorkday: false,
+  ...overrides,
+})
```
Comment (dmaklygin) [inline src/calendar/__specs__/day_spec.js]: 👍

File: src/calendar/__specs__/month_header_spec.js [hunk 9]
```diff
@@ -1,62 +1,52 @@
...
   describe('#_switchMonth', () => {
     it('calls #onMonthChange', () => {
-      props = Object.assign(defaultProps, {
-        maxDate: new Date(2015, 7, 1),
-        onMonthChange: jest.fn(),
-      })
-      wrapper = shallow(<MonthHeader {...props} />)
-      wrapper.instance()._switchMonth(1)
+      wrapper.setProps({maxDate: new Date(2015, 7, 1)})
+      instance._switchMonth(1)
```
Comment (dmaklygin) [inline src/calendar/__specs__/month_header_spec.js]: let's add `toBeCalledTimes(1)`

File: src/calendar/day.jsx [hunk 10]
```diff
@@ -1,53 +1,69 @@
...
+  isSelectable: PropTypes.bool.isRequired,
+  isSelected: PropTypes.bool.isRequired,
+  isSelectionEnd: PropTypes.bool.isRequired,
+  isSelectionStart: PropTypes.bool.isRequired,
+  isToday: PropTypes.bool.isRequired,
+  isWeekend: PropTypes.bool.isRequired,
+  isWorkday: PropTypes.bool.isRequired,
 }
+
+export default Day
```
Comment (dmaklygin) [inline src/calendar/day.jsx]: 👍

File: src/calendar/__specs__/month_header_spec.js (hunk 9 above)
Comment (PeterMK85) [reply src/calendar/__specs__/month_header_spec.js]: Added

**Pull request merged**